lighthouse-core/audits/preload-lcp.js

```diff
@@ -0,0 +1,185 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const URL = require('../lib/url-shim.js');
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+const MainResource = require('../computed/main-resource.js');
+const LanternLCP = require('../computed/metrics/lantern-largest-contentful-paint.js');
+const LoadSimulator = require('../computed/load-simulator.js');
+const UnusedBytes = require('./byte-efficiency/byte-efficiency-audit.js');
+
+const UIStrings = {
+  /** Title of a lighthouse audit that tells a user to preload an image in order to improve their LCP time. */
+  title: 'Preload Largest Contentful Paint image',
+  /** Description of a lighthouse audit that tells a user to preload an image in order to improve their LCP time.  */
+  description: 'Consider preloading the image used by ' +
+    'the LCP element in order to improve your LCP time.',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class PreloadLCPAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'preload-lcp',
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs', 'URL', 'TraceElements'],
+      scoreDisplayMode: Audit.SCORING_MODES.NUMERIC,
+    };
+  }
+
+  /**
+   *
+   * @param {LH.Artifacts.NetworkRequest} request
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {Array<LH.Gatherer.Simulation.GraphNode>} initiatorPath
+   * @param {string|undefined} lcpImageSource
+   * @return {boolean}
+   */
+  static shouldPreloadRequest(request, mainResource, initiatorPath, lcpImageSource) {
+    const mainResourceDepth = mainResource.redirects ? mainResource.redirects.length : 0;
+
+    // If it's not the request for the LCP image, don't recommend it.
+    if (request.url !== lcpImageSource) return false;
+    // If it's already preloaded, no need to recommend it.
+    if (request.isLinkPreload) return false;
+    // It's not a request loaded over the network, don't recommend it.
+    if (URL.NON_NETWORK_PROTOCOLS.includes(request.protocol)) return false;
+    // It's not at the right depth, don't recommend it.
+    if (initiatorPath.length < mainResourceDepth + 1) return false;
+    // Finally, return whether or not it belongs to the main frame
+    return request.frameId === mainResource.frameId;
+  }
+
+  /**
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @param {LH.Artifacts.TraceElement|undefined} lcpElement
+   * @return {string|null}
```

Comment:
nit: we try to avoid `null` unless there's a real need (like differentiating for definedness in the config defaults)
```suggestion
   * @return {string|undefined}
```

same deal will apply to all `null` uses in this file I think?